lib/fluent/plugin/out_forward.rb

```diff
@@ -303,23 +312,25 @@ def send_data(node, tag, chunk)
         # writeRaw(tag)
         sock.write tag.to_msgpack  # tag
 
-        # beginRaw(size)
-        sz = chunk.size
-        #if sz < 32
-        #  # FixRaw
-        #  sock.write [0xa0 | sz].pack('C')
-        #elsif sz < 65536
-        #  # raw 16
-        #  sock.write [0xda, sz].pack('Cn')
-        #else
-        # raw 32
-        sock.write [0xdb, sz].pack('CN')
-        #end
-
-        # writeRawBody(packed_es)
-        chunk.write_to(sock)
-
-        option = { 'size' => chunk.size_of_events }
+        chunk.open(compressed: @compress) do |chunk_io|
+          # beginRaw(size)
+          sz = chunk_io.size
+
+          #if sz < 32
+          #  # FixRaw
+          #  sock.write [0xa0 | sz].pack('C')
+          #elsif sz < 65536
+          #  # raw 16
+          #  sock.write [0xda, sz].pack('Cn')
+          #else
+          # raw 32
+          sock.write [0xdb, sz].pack('CN')
+
+          # writeRawBody(packed_es)
+          IO.copy_stream(chunk_io, sock)
+        end
+
+        option = { 'size' => chunk.size_of_events, 'compress' => @compress }
```

Comment:
And, the value should be a string, not symbol. Do `@compress.to_s` to do it.
The default representation of Symbol in msgpack-ruby is `str`, but it can be overridden. So make it sure to feed String as value.